libraries/AdaptiveExpressions/ExpressionFunctions.cs

```diff
@@ -1626,10 +1633,10 @@ private static (object value, string error) Where(Expression expression, IMemory
 
                         // the local iterator is pushed as one memory layer in the memory stack
                         stackedMemory.Push(new SimpleObjectMemory(local));
-                        var (r, _) = expression.Children[2].TryEvaluate<bool>(stackedMemory);
+                        var (r, _) = expression.Children[2].TryEvaluate(stackedMemory, new Options(options) { NullSubstitution = null });
```

Comment:
What happens if you get an error here?

Reply:
the r would be null, usually, but good catch, i now catch error specificly